### PR TITLE
chore: standardize the simple cache client tsdocs

### PR DESCRIPTION
**Pull request by nand4011**
Standardize the tsdocs in the simple cache client. Formats them in a consistent way and uses consistent language for descriptions and matching parameters.

I'm not attached to any of the specific language so if we have preferred wording I can update this.

### CODE REVIEW

**Review by cprice404 — APPROVED** [review]
wow

thank you!

File: src/simple-cache-client.ts [hunk 1]
```diff
@@ -232,9 +239,12 @@ export class SimpleCacheClient {
...
 
   /**
-   * Get the number of values in a list.
-   * @param {string} cacheName - Name of the cache with the list.
+   * Gets the number of elements in the given list.
+   *
+   * @param {string} cacheName - The cache containing the list.
    * @param {string} listName - The list to get the length of.
+   * @returns {Promise<CacheListLength.Response>} - The result of the get
+   * operation (hit, miss, or error) and the list length.
```
Comment (malandis) [inline src/simple-cache-client.ts:247]: The wording here trips me up a bit since the list length is only part of a hit response. Can we rephrase this to make that clear? (Same goes for other operations that carry extra information in the hit/success)

File: src/simple-cache-client.ts [hunk 2]
```diff
@@ -259,10 +271,12 @@ export class SimpleCacheClient {
...
   /**
-   * Get and remove the first value from a list.
-   * @param {string} cacheName - Name of the cache with the list.
+   * Gets and removes the first value from the given list.
+   *
+   * @param {string} cacheName - The cache containing the list.
    * @param {string} listName - The list to pop.
-   * @returns {Promise<CacheListPopFront.Response>}
+   * @returns {Promise<CacheListPopFront.Response>} - The result of the pop
+   * operation (hit, miss, or error) and the removed element.
```
Comment (malandis) [inline src/simple-cache-client.ts:279]: Same here

File: src/simple-cache-client.ts [hunk 3]
```diff
@@ -104,12 +104,12 @@ export class SimpleCacheClient {
...
-   * @param {(string | Uint8Array)} key - The key to lookup.
-   * @returns {Promise<CacheGet.Response>} - Promise containing the status
-   * of the get operation (hit or miss) and the associated value.
-   * @memberof SimpleCacheClient
+   * Gets the value stored for the given key.
+   *
+   * @param {string} cacheName - The cache to perform the lookup in.
+   * @param {string | Uint8Array} key - The key to look up.
+   * @returns {Promise<CacheGet.Response>} - The result of the get operation
+   * (hit, miss, or error) and the value.
```
Comment (malandis) [inline src/simple-cache-client.ts:112]: And here

File: src/simple-cache-client.ts [hunk 4]
```diff
@@ -218,10 +223,12 @@ export class SimpleCacheClient {
...
   /**
-   * Fetch the entire list.
-   * @param {string} cacheName - Name of the cache to fetch the list from.
+   * Fetches all elements of the given list.
+   *
+   * @param {string} cacheName - The cache containing the list.
    * @param {string} listName - The list to fetch.
-   * @returns {Promise<CacheListFetch.Response>}
+   * @returns {Promise<CacheListFetch.Response>} - The result of the fetch
+   * operation (hit, miss, or error) and the list elements.
```
Comment (malandis) [inline src/simple-cache-client.ts:231]: Here

File: src/simple-cache-client.ts [hunk 5]
```diff
@@ -245,10 +255,12 @@ export class SimpleCacheClient {
...
   /**
-   * Get and remove the last value from a list.
-   * @param {string} cacheName - Name of the cache with the list.
+   * Gets and removes the last value from the given list.
+   *
+   * @param {string} cacheName - The cache containing the list.
    * @param {string} listName - The list to pop.
-   * @returns {Promise<CacheListPopBack.Response>}
+   * @returns {Promise<CacheListPopBack.Response>} - The result of the pop
+   * operation (hit, miss, or error) and the removed element.
```
Comment (malandis) [inline src/simple-cache-client.ts:263]: Here

File: src/simple-cache-client.ts [hunk 6]
```diff
@@ -345,11 +369,12 @@ export class SimpleCacheClient {
...
-   * Fetch the entire set from the cache.
-   * @param {string} cacheName - Name of the cache to perform the lookup in.
+   * Fetches all elements of the given set
+   *
+   * @param {string} cacheName - The cache containing the set.
    * @param {string} setName - The set to fetch.
-   * @returns Promise<SetFetch.Response> - Promise containing the result of the fetch operation and the associated set.
-   * @memberof SimpleCacheClient
+   * @returns {Promise<CacheSetFetch.Response>} - The result of the fetch
+   * operation (hit, miss, or error) and the set elements.
```
Comment (malandis) [inline src/simple-cache-client.ts:377]: Here

File: src/simple-cache-client.ts [hunk 7]
```diff
@@ -450,41 +489,44 @@ export class SimpleCacheClient {
...
-   * Fetch the entire dictionary from the cache.
-   * @param {string} cacheName - Name of the cache to perform the lookup in.
+   * Fetches all elements of the given dictionary.
+   *
+   * @param {string} cacheName - The cache to perform the lookup in.
    * @param {string} dictionaryName - The dictionary to fetch.
-   * @returns {Promise<CacheDictionaryFetch.Response>}- Promise containing the result of the fetch operation and the associated dictionary.
-   * @memberof SimpleCacheClient
+   * @returns {Promise<CacheDictionaryFetch.Response>} - The result of the fetch
+   * operation (hit, miss, or error) and the dictionary elements.
```
Comment (malandis) [inline src/simple-cache-client.ts:529]: Here

File: src/simple-cache-client.ts [hunk 8]
```diff
@@ -569,11 +619,13 @@ export class SimpleCacheClient {
...
-   * @param {string} cacheName - Name of the cache to perform the lookup in.
+   * Gets multiple values from the given dictionary.
+   *
+   * @param {string} cacheName - The cache containing the dictionary.
    * @param {string} dictionaryName - The dictionary to look up.
-   * @param {string[] | Uint8Array[]} fields - The field in the dictionary to lookup.
-   * @returns {Promise<CacheDictionaryGetField>}- Promise containing the status and associated value for each field.
+   * @param {string[] | Uint8Array[]} fields - The fields to look up.
+   * @returns {Promise<CacheDictionaryGetFields.Response>} - The result of the
+   * get operation (hit, miss, or error) and the elements.
```
Comment (malandis) [inline src/simple-cache-client.ts:628]: Here

**Review by malandis — COMMENTED** [review]
Thanks for this!

One observation about the wording in the documentation. Feel free to override as I do not feel too strongly about it, but it did jump out

File: src/simple-cache-client.ts [hunk 9]
```diff
@@ -331,8 +347,9 @@ export class SimpleCacheClient {
    * @param {CollectionTtl} [options.ttl] - How the TTL should be managed.
    * Refreshes the list's TTL using the client's default if this is not
    * supplied.
-   * @returns {Promise<CacheListPushFront.Response>} - The result of the push
-   * operation (success or error).
+   * @returns {Promise<CacheListPushFront.Response>} -
+   * {@link CacheListPushFront.Success} on success.
```
Comment (malandis) [inline src/simple-cache-client.ts:351]: Success also contains the list length

File: src/simple-cache-client.ts [hunk 10]
```diff
@@ -299,8 +314,9 @@ export class SimpleCacheClient {
    * @param {CollectionTtl} [options.ttl] - How the TTL should be managed.
    * Refreshes the list's TTL using the client's default if this is not
    * supplied.
-   * @returns {Promise<CacheListPushBack.Response>} - The result of the push
-   * operation (success or error).
+   * @returns {Promise<CacheListPushBack.Response>} -
+   * {@link CacheListPushBack.Success} on success.
```
Comment (malandis) [inline src/simple-cache-client.ts:318]: Success also contains the list length

File: src/simple-cache-client.ts [hunk 11]
```diff
@@ -700,8 +747,9 @@ export class SimpleCacheClient {
    * @param {CollectionTtl} [options.ttl] - How the TTL should be managed.
    * Refreshes the dictionary's TTL using the client's default if this is not
    * supplied.
-   * @returns {Promise<CacheDictionaryIncrement>} - The result of the increment
-   * operation (success or error).
+   * @returns {Promise<CacheDictionaryIncrement>} -
+   * {@link CacheDictionaryIncrement.Success} on success.
```
Comment (malandis) [inline src/simple-cache-client.ts:751]: Also contains the value post-increment

File: src/simple-cache-client.ts [hunk 12]
```diff
@@ -700,8 +747,9 @@ export class SimpleCacheClient {
...
+   * {@link CacheDictionaryIncrement.Error} on failure.
```
Comment (malandis) [inline src/simple-cache-client.ts:752]: Important failure mode to document: the value to be incremented was not an integer or string representation of an integer

**Review by malandis — CHANGES_REQUESTED** [review]
This looks great. Just a couple more changes for completeness along the lines of what you've done already :+1:
